app/api/groups/remove_user_batch.go

```diff
@@ -0,0 +1,108 @@
+package groups
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/loginmodule"
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:operation DELETE /user-batches/{group_prefix}/{custom_prefix} groups userBatchRemove
+// ---
+// summary: Remove a user batch
+// description:
+//   Lets an admin remove user batches and all users having "{group_prefix}_{custom_prefix}_" as login prefix.
+//
+//   On success the service
+//
+//     * requests the login module to delete the users with {group_prefix}_{custom_prefix}_" as prefix
+//       (/platform_api/accounts_manager/delete with prefix parameter);
+//
+//     * deletes all users with "{group_prefix}_{custom_prefix}_" as prefix
+//       (ignoring the membership locks on groups the authenticated user manage (but not others!));
+//
+//     * deletes the user batch entry.
+//
+//
+//   The authenticated user should be a manager of the `group_prefix`'s group (or its ancestor)
+//   with `can_manage` >= 'memberships', otherwise the 'forbidden' error is returned.
+//
+//
+//   If there are users with locked membership in groups the current user cannot manage,
+//   the 'unprocessable entity' error is returned.
+// parameters:
+// - name: group_prefix
+//   in: path
+//   type: string
+//   required: true
+// - name: custom_prefix
+//   in: path
+//   type: string
+//   required: true
+// responses:
+//   "200":
+//     "$ref": "#/responses/deletedResponse"
+//   "401":
+//     "$ref": "#/responses/unauthorizedResponse"
+//   "403":
+//     "$ref": "#/responses/forbiddenResponse"
+//   "422":
+//     "$ref": "#/responses/unprocessableEntityResponse"
+//   "500":
+//     "$ref": "#/responses/internalErrorResponse"
+func (srv *Service) removeUserBatch(w http.ResponseWriter, r *http.Request) service.APIError {
+	groupPrefix := chi.URLParam(r, "group_prefix")
+	customPrefix := chi.URLParam(r, "custom_prefix")
+
+	user := srv.GetUser(r)
+	managedByUser := srv.Store.ActiveGroupAncestors().ManagedByUser(user).
+		Where("can_manage != 'none'").
+		Select("groups_ancestors_active.child_group_id AS id")
+
+	// The user batch should exist and the current user should be a manager of the group
+	// linked to the group_prefix
+	found, err := srv.Store.UserBatches().
+		Joins("JOIN user_batch_prefixes USING(group_prefix)").
+		Where("group_prefix = ?", groupPrefix).
+		Where("custom_prefix = ?", customPrefix).
+		Where("user_batch_prefixes.group_id IN (?)", managedByUser.QueryExpr()).
```

Comment:
wouldn't just joining do the job and be more efficient?

Reply:
fixed